x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -560,6 +562,9 @@ private BucketExtractor createExtractor(FieldExtraction ref, BucketExtractor tot
             }
 
             if (ref == GlobalCountRef.INSTANCE) {
+                if (totalCount == null) {
```

Comment:
Not sure this is safety check however right now it looks as if we're not sure if this change will not backfire.

Reply:
This is probably an excess of defensive programming, looking at the code this case should never happen.
Anyway, with the suggested changes this check can be removed.